scripts/release-notes/generate-release-notes.ts

```diff
@@ -0,0 +1,483 @@
+#!/usr/bin/env tsx
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { Octokit } from '@octokit/rest';
+import { components } from '@octokit/openapi-types';
+import * as fs from 'fs';
+import mustache from 'mustache';
+
+type Issue = components['schemas']['issue'];
+type Milestone = components['schemas']['milestone'];
+
+interface HighlitedPR {
+  title: string;
+  shortDesc: string;
+  longDesc: string;
+}
+
+interface Author {
+  username: string;
+  link: string;
+}
+
+interface PRInfo {
+  title: string;
+  author: Author;
+  number: number;
+  link: string;
+  created_at?: string;
+}
+
+interface PRCategory {
+  category: string;
+  prs: PRInfo[];
+}
+
+export class ReleaseNotesPreparator {
+  private octokit;
+  constructor(
+    private token: string,
+    private organization: string,
+    private repo: string,
+    private milestone: string,
+    private username: string,
+    private model: string | undefined,
+    private port: string,
+    private endpoint: string,
+    private useOllama: boolean,
+  ) {
+    this.octokit = new Octokit({ auth: token });
+  }
+
+  private async generateMD(
+    changelog: PRCategory[],
+    firstTimeContributors: PRInfo[],
+    highlighted: HighlitedPR[],
+  ): Promise<void> {
+    const date = new Date();
+    const formattedDate = date.toISOString().split('T')[0];
+    const version = this.milestone.slice(0, -2);
+    const releaseNotesTemplate = fs.readFileSync('./scripts/release-notes/release-notes.mustache', 'utf8');
```

Comment:
as in async mode, use the await readFile version ? (you're using it few lines later to write a file fs.promises.writeFile)

or import 'node:fs/promises' instead